tests/test_polygon.py

```diff
@@ -109,13 +109,23 @@ def test_numpy_linearring_coords():
     assert_array_equal(ra, expected)
 
 
+@pytest.mark.filterwarnings("error:An exception was ignored")  # NumPy 1.21
 def test_numpy_empty_linearring_coords():
     np = pytest.importorskip("numpy")
 
     ring = LinearRing()
     assert np.asarray(ring.coords).shape == (0,)
 
 
+@pytest.mark.filterwarnings("error:An exception was ignored")  # NumPy 1.21
```

Comment:
This is the only test that didn't need `@shapely20_deprecated` to silence "ShapelyDeprecationWarning: The array interface is deprecated...", and I have no idea why. @jorisvandenbossche ?

Reply:
So I _think_ this is because Polygon already didn't have an array interface before (its rings have one), so for polygon the array interface didn't need to be deprecated. In addition, it also doesn't have deprecation of `__len__` or iteration like the multi-part geometries (that also didn't work before), where those warnings are also typically raises when interacting with numpy.